naff/models/discord/channel.py

```diff
@@ -142,7 +142,7 @@ async def fetch(self) -> list["GuildForumPost"]:
             expected = self.get_limit
 
             rcv = await self.channel._client.http.list_public_archived_threads(
-                self.channel.id, limit=expected, before=self.last
+                self.channel.id, limit=expected, before=self.last.id
```

Comment:
This will throw an attribute error if the user passes an ID to the Async Iterator - something which is common with iterators. Suggest using `to_snowflake` instead

Reply:
Fair point, fixed.